src/test/transaction_tests.cpp

```diff
@@ -152,14 +152,16 @@ unsigned int FillFlags(unsigned int flags)
     return flags;
```

Comment:
unrelated nit: Could `Assert(IsValidFlagCombination(flags));`? (Needs #21553)

Reply:
There are places where we need to fill the flags as well as check that they're a valid combination

Reply:
I completely misread this for some reason 🤦 sorry about that
Just rebased on top of #21553 and added the asserts